src/user/user.controller.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpException,
+  HttpStatus,
+  Post,
+} from '@nestjs/common';
+import { UserService } from './user.service';
+
+@Controller('user')
+export class UserController {
+  constructor(private userservice: UserService) {}
+
+  @Post('create')
+  createNewUser(
+    @Body() userDto: { id: number; username: string; password: string },
+  ) {
+    return this.userservice.createUser(
+      userDto.id,
+      userDto.username,
+      userDto.password,
+    );
+  }
+  @Get('getusers')
+  async findAllUsers() {
+    try {
+      const users = await this.userservice.getAllUsers();
+      return users;
+    } catch (error) {
+      throw new HttpException(
+        'error from controller',
+        HttpStatus.INTERNAL_SERVER_ERROR,
```

Comment:
Its not a 500 error, this should be handled in an error propagation module , let us talk about it in the next meeting, but overall u should send a meaning full error message here